src/main/scala/diplomacy/BundleBridge.scala

```diff
@@ -56,8 +61,12 @@ case class BundleBridgeSource[T <: Data](genOpt: Option[() => T] = None)(implici
 {
   def bundle: T = out(0)._1
 
+  private def inferInput = bundle.getElements.forall { elt =>
+    DataMirror.directionOf(elt) == ActualDirection.Unspecified
+  }
+
   def makeIO()(implicit valName: ValName): T = {
-    val io: T = IO(Flipped(chiselTypeClone(bundle)))
+    val io: T = IO(if (inferInput) Input(chiselTypeOf(bundle)) else Flipped(chiselTypeClone(bundle)))
     io.suggestName(valName.name)
     bundle <> io
```

Comment:
I just felt confused here.  By code reading, I felt like the direction of makeIO of BundleBridgeSourceNode is Input, While the direction of makeIO of BundleBridgeSinkNode is Output. I felt like this is strange, the source node only has output bundles and the sink only gets input right? Why this is opposite here. Also,  if inferInput is false in the makeIO of BundleBridgeSource, why we need to Flipped the orignal direction, while inside the BundleBridgeSink, there is no Flipped?  it is wired. Seems like  the io after calling makeIO in corresponding node is opposite with its original IO, why? 
@hcook @mwachs5 @jackkoenig

Reply:
> I just felt confused here. By code reading, I felt like the direction of makeIO of BundleBridgeSourceNode is Input, While the direction of makeIO of BundleBridgeSinkNode is Output. I felt like this is strange, the source node only has output bundles and the sink only has input right? Why this is opposite here. Also, if inferInput is false in the makeIO of BundleBridgeSource, why we need to Flipped the orignal direction, while inside the BundleBridgeSink, there is no Flipped? it is wired. Seems like the io after calling makeIO in corresponding node is opposite with its original IO, why?
> @hcook @mwachs5 @jackkoenig

I answer (more or less) this question in stackoverflow https://stackoverflow.com/questions/66963837/how-to-understand-the-flip-in-autobundle-and-in-makeios